providers/marketo/read_test.go

```diff
@@ -0,0 +1,101 @@
+package marketo
+
+import (
+	"errors"
+	"net/http"
+	"testing"
+
+	"github.com/amp-labs/connectors"
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/internal/datautils"
+	"github.com/amp-labs/connectors/test/utils/mockutils/mockserver"
+	"github.com/amp-labs/connectors/test/utils/testroutines"
+	"github.com/amp-labs/connectors/test/utils/testutils"
+)
+
+func TestRead(t *testing.T) { // nolint:funlen,gocognit,cyclop
+	t.Parallel()
+
+	zeroRecords := testutils.DataFromFile(t, "zero-records.json")
+	unsupportedResponse := testutils.DataFromFile(t, "not-found.json")
+	campaignsResponse := testutils.DataFromFile(t, "campaigns.json")
+
+	tests := []testroutines.Read{
+		{
+			Name:         "Object Name is required",
+			Server:       mockserver.Dummy(),
+			ExpectedErrs: []error{common.ErrMissingObjects},
+		},
+		{
+			Name:         "At least one field is required",
+			Input:        common.ReadParams{ObjectName: "deals"},
+			Server:       mockserver.Dummy(),
+			ExpectedErrs: []error{common.ErrMissingFields},
+		},
+		{
+			Name:  "Unsupported object",
+			Input: common.ReadParams{ObjectName: "arsenal", Fields: datautils.NewStringSet("testField")},
+			Server: mockserver.Fixed{
+				Setup:  mockserver.ContentJSON(),
+				Always: mockserver.ResponseString(http.StatusBadRequest, string(unsupportedResponse)),
```

Comment:
```suggestion
				Always: mockserver.Response(http.StatusBadRequest, unsupportedResponse),
```
It is already of `[]byte` type. True for other places in the PR.